dl/Util.py

```diff
@@ -316,6 +316,40 @@ def split_auth_token(token):
     return res.groups() if res else None
 
 
+def auth_token2dict(token):
+    """
+    Given an auth token splits it in its components
+    and returns a dictionary.
+    Parameters
+    ----------
+    token : str
+        A string auth token
+        E.g.
+        "testuser.3666.3666.$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/"
+
+    Returns
+    -------
+    return:
+           { 'username': "username value",
+              'uid': "numeric user id",
+              'gid': "numeric group id",
+              'hash': hash
+            }
+    E.g.   { 'username': "testuser",
+              'uid': "3666",
+              'gid': "3666",
+              'hash': "$1$PKCFmMzy$OPpZg/ThBmZe/V8LVPvpi/"
+            }
+            or None if token not valid
+    """
```

Comment:
Consider adding tests for this function in `./tests/test_util.py` (assuming there is time)

Reply:
yes, I could add something! thank you for reminding me!